packages/blockchain/src/db/operation.ts

```diff
@@ -144,9 +144,10 @@ export class DBOp {
   }
 
   public updateCache(cacheMap: CacheMap) {
-    if (this.cacheString && cacheMap[this.cacheString] && Buffer.isBuffer(this.baseDBOp.value)) {
+    if (this.cacheString && cacheMap[this.cacheString]) {
       if (this.baseDBOp.type == 'put') {
-        cacheMap[this.cacheString].set(this.baseDBOp.key, this.baseDBOp.value)
+        Buffer.isBuffer(this.baseDBOp.value) &&
```

Comment:
thanks for the PR!

i find this type of conditional chaining kind of awkward, can you instead add to the if statement above?

Reply:
```ts
public updateCache(cacheMap: CacheMap) {
  if (this.cacheString && cacheMap[this.cacheString]) {
    if (this.baseDBOp.type == 'put' && Buffer.isBuffer(this.baseDBOp.value)) {
      cacheMap[this.cacheString].set(this.baseDBOp.key, this.baseDBOp.value)
    } else if (this.baseDBOp.type == 'del') {
      cacheMap[this.cacheString].del(this.baseDBOp.key)
    } else {
      throw new Error('unsupported db operation on cache')
    }
  }
}
```

It will throw an exception if type is `put` and value is not `Buffer`

Reply:
Ah ok, thanks for the explanation! 👍

Reply:
gotcha, thanks. i think i still would've preferred if we formatted this as an additional inside if statement

```
if (this.baseDBOp.type == 'put') {
  if (Buffer.isBuffer(this.baseDBOp.value)) {
    cacheMap[this.cacheString].set(this.baseDBOp.key, this.baseDBOp.value)
  }
}
```

but that's fine for now. thanks again for the pr :)

good point about missing tests, would be great to have some for these

Reply:
Ok, guess we can maybe update "on occasion" if someone stumbles upon further down the road (e.g. @emersonmacro along his work on #1044).